api/internal/localizer/localizer.go

```diff
@@ -141,8 +132,47 @@ func (lc *localizer) localizeNativeFields(kust *types.Kustomization) error {
 		}
 	}
 
-	// TODO(annasong): localize all other kustomization fields: resources, bases, crds, configurations,
-	// openapi, patchesJson6902, patchesStrategicMerge, replacements
+	for name, patches := range map[string][]types.Patch{
```

Comment:
Instead of this double for-loop/map structure, a more common and maybe simpler pattern is to write a function, for example:

```
    if err := localizePatches("patches", kust.Patches); err != nil {
      return err
    }
    if err := localizePatches("patchesJson6902", kust.PatchesJson6902); err != nil {
      return err
    }

...

func localizePatches(name, patches []Patch) error {
    for i := range patches {
	    if patches[i].Path != "" {
		    newPath, err := lc.localizeFile(patches[i].Path)
		    if err != nil {
			    return errors.WrapPrefixf(err, "unable to localize %s path %q", name, patches[i].Path)
		    }
		    patches[i].Path = newPath
	    }
    }
}
```

This is a similar structure to how you did the ConfigMapGenerator/SecretGenerator. We can consider looping through a map when there are more elements, but for just these two fields IMO just calling a common function is easier to read and reason about what is happening.

Reply:
Done. Thanks for the suggestion and code!